packages/theme-check-common/src/checks/valid-preset-and-default-settings/index.ts

```diff
@@ -0,0 +1,272 @@
+import { isSection, isBlock } from '../../to-schema';
+import { basename } from '../../path';
+import {
+  JSONNode,
+  LiquidCheckDefinition,
+  LiteralNode,
+  Severity,
+  SourceCodeType,
+} from '../../types';
+import { nodeAtPath } from '../../json';
+import {
+  BlockNodeWithPath,
+  getBlocks,
+  validateBlockFileExistence,
+} from '../valid-block-target/block-utils';
+import { Section } from '../../types/schemas/section';
+
+export const ValidPresetAndDefaultSettings: LiquidCheckDefinition = {
+  meta: {
+    code: 'ValidPresetAndDefaultSettings',
+    name: 'Reports invalid preset and default settings for sections and blocks',
+    docs: {
+      description: 'Reports invalid preset and default settings for sections and blocks',
+      recommended: true,
+      url: 'https://shopify.dev/docs/storefronts/themes/tools/theme-check/checks/valid-preset-and-default-settings',
+    },
+    severity: Severity.ERROR,
+    type: SourceCodeType.LiquidHtml,
+    schema: {},
+    targets: [],
+  },
+
+  create(context) {
+    function getSchema() {
+      const name = basename(context.file.uri, '.liquid');
+      switch (true) {
+        case isBlock(context.file.uri):
+          return context.getBlockSchema?.(name);
+        case isSection(context.file.uri):
+          return context.getSectionSchema?.(name);
+        default:
+          return undefined;
+      }
+    }
+
+    const extractDefaultBlockSettingIds = (
+      ast: JSONNode,
+      blocks: Record<string, any>[],
+      path: string[] = ['default', 'blocks'],
+    ): { key: string; start: any; end: any }[] => {
+      let settingIds: { key: string; start: any; end: any }[] = [];
+
+      for (const [blockIndex, block] of blocks.entries()) {
+        const currentPath = [...path, blockIndex.toString()];
+
+        for (const [settingId, _] of Object.entries(block.settings ?? {})) {
+          const node = nodeAtPath(ast, [...currentPath, 'settings']);
```

Comment:
Need to nest this one level deeper - the node is highlighting all of settings rather than the specific id key